### ISSUE TITLE
Tagging a release including the async APIs

### ISSUE THREAD

**Issue by SylvainCorlay** [jupyter/jupyter_client#511]
Hey, we are blocking on the jupyter_client async API being included in a release (pun unintended).

Is there any objection to us tagging a jupyter_client release in the next couple of days?

cc @minrk @Carreau @jasongrout

**Comment by MSeal** [comment]
If it's against the 5.x branch it's probably good but the master branch is way out of sync if the change is PRd against that. I don't have the context on what's the state / intention with master but releases and code changes for the past year or two have been out of that branch. Perhaps one of the pinged people can assist or help with merging 5.x back into master?

**Comment by SylvainCorlay** [comment]
This change is in the master branch.

Do you mean that there are 5.x PRs that are not backports of PRs of the master branch?

**Comment by MSeal** [comment]
Yes. 5.x is deviated far from master and has critical changes in it that we should not make a release without.

**Comment by SylvainCorlay** [comment]
> Yes. 5.x is deviated far from master and has critical changes in it that we should not make a release without.

This is *very* unfortunate. I think that we should not have done this in the 5.x branch.

We need to forward port these changes to master.

**Comment by MSeal** [comment]
I've asked on a few threads why it is that way but the folks who made that decision / did the work haven't been active in the past year.

There are some non-trivial changes in master that I don't know the impact of, nor if they were tested against existing downstream libraries. Someone likely needs to evaluate a 6.0 release candidate for these things with a prerelease. I've been managing the nbconvert 6 and papermill 2 releases so it's been pretty low on my list of things to dig into.

**Comment by SylvainCorlay** [comment]
OK, thanks. I just saw your comment in the async API PR about this.

I will look into it in the next week or so.

**Comment by davidbrochart** [comment]
I could try and merge 5.x back into master.

**Comment by davidbrochart** [comment]
What would be the best strategy here, rebase 5.x onto master, merge master into 5.x?

**Comment by davidbrochart** [comment]
I merged `5.x` back into `master` in https://github.com/jupyter/jupyter_client/pull/513. Could this be merged, or do you think more tests are needed?

**Comment by MSeal** [comment]
Thanks @davidbrochart for all the effort here. One of us is going to need to dig into what had changed in master and make sure it's compatible to the downstream clients. I can do some more thorough testing and code analysis next week to that effort. A brief scan of all the commits in the last 2 years looks like nothing unreasonable.

**Comment by SylvainCorlay** [comment]
@MSeal now that we are back to a single reference branch, I propose that we get back to pushing towards a release.

Should we make it a 6.0 (with a release candidate) or another 5.x?

**Comment by MSeal** [comment]
I looked through every commit involved and did testing on a number of possible edge cases I know had been problems in the past. With downstream library test coverage being good and the changes not looking all that scary with the experimental change moved out of master, I opted to just release 6.0.0: https://discourse.jupyter.org/t/jupyter-client-6-0-0/3414

I also removed all of the obvious python 2 lines that could be safely removed without any logic changes. There's more we could easily rip out but I got lazy and wanted to only make changes that I felt safe making for the release without code review.
